### PR TITLE
Allow hard-disconnect

### PR DESCRIPTION
**Pull request by wraitii**
When calling disconnect, give the option to forget the default/last wallet, so that new calls to connect are reset to a default state.

--

I was doing this on briq before and I think it's a worthwhile API change, if only to be able to test a fresh 'connect' easily.

### CODE REVIEW

File: dist/index.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import type { GetStarknetWalletOptions, IStarknetWindowObject } from "./types";
-export declare const getStarknet: () => IStarknetWindowObject;
+export declare const getStarknet: (resetAuthorizations?: boolean | undefined) => IStarknetWindowObject;
```
Comment (janek26) [inline dist/index.d.ts:2]: lets just keep it in one place :) I think `disconnect` is the best place for it

File: dist/index.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 export declare const connect: (options?: GetStarknetWalletOptions | undefined) => Promise<IStarknetWindowObject | undefined>;
-export declare const disconnect: () => boolean;
+export declare const disconnect: (resetAuthorizations?: boolean | undefined) => boolean;
```
Comment (janek26) [inline dist/index.d.ts:4]:
```suggestion
export declare const disconnect: (resetAuthorizations?: boolean) => boolean;
```
not needed, as it's optional anyways (`?`)

File: src/index.ts [hunk 3]
```diff
@@ -89,16 +89,23 @@ class GetStarknetWallet implements IGetStarknetWallet {
...
+            defaultWallet.delete();
+            lastWallet.delete();
+        }
+
         // disconnected successfully if was connected before
         return connected;
     };
 
-    getStarknet = (): IStarknetWindowObject => {
+    getStarknet = (resetAuthorizations?: boolean): IStarknetWindowObject => {
```
Comment (janek26) [inline src/index.ts:108]: remove pls

**Review by janek26 — CHANGES_REQUESTED** [review]

File: dist/index.d.ts (hunk 2 above)
Comment (wraitii) [reply dist/index.d.ts]: Seems to have been added by `npm run build`.

Comment (janek26) [reply dist/index.d.ts]: oh! build files will (hopefully) be excluded soon anyways! :D

**Review by janek26 — APPROVED** [review]

**Pull request merged**